lib/fencing/st_client.c

```diff
@@ -919,27 +919,29 @@ invoke_registered_callbacks(stonith_t *stonith, xmlNode *msg, int call_id)
     }
 
     if ((cb_info != NULL) && (cb_info->callback != NULL)
-        && (rc == pcmk_ok || !(cb_info->only_success))) {
+        && (pcmk__result_ok(&result) || !(cb_info->only_success))) {
         crm_trace("Invoking callback %s for call %d",
                   crm_str(cb_info->id), call_id);
-        invoke_fence_action_callback(stonith, call_id, rc, cb_info->user_data,
-                                     cb_info->callback);
+        invoke_fence_action_callback(stonith, call_id, &result,
+                                     cb_info->user_data, cb_info->callback);
 
-    } else if ((private->op_callback == NULL) && (rc != pcmk_ok)) {
-        crm_warn("Fencing action without registered callback failed: %s",
-                 pcmk_strerror(rc));
+    } else if ((private->op_callback == NULL) && !pcmk__result_ok(&result)) {
+        crm_warn("Fencing action without registered callback failed: %d (%s)",
+                 result.exit_status,
+                 pcmk_exec_status_str(result.execution_status));
```

Comment:
now that we have it couldn't we add the textual failure-result?

Reply:
This can't happen internally so I wasn't too worried about it, but you're right, it should